### ISSUE TITLE
UI test B44476 needs to be automated

### ISSUE THREAD

**Issue by samhouts** [xamarin/Xamarin.Forms#2381]
- [ ] Test needs to be automated

**Comment by AndreiMisiukevich** [comment]
@samhouts hi
Could you please explain, how to run UI Test on simulator
For example run test only for one this issue 

https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3840.cs#L80-L90

I found, how to open it as part of App, but how to test **TranslatingViewKeepsScrollViewPosition**

IDE: Visual Studio for Mac

Thanks

**Comment by puppetSpace** [comment]
Gonna try this as my first contribution.

**Comment by puppetSpace** [comment]
I have to say. It took a while before my environment was ready. There are some things missing in the "getting started" section. Especially for running UITest.

First problem was that the test weren't discovered.
- Either install older version of nunit test adapter extension
- Or change a setting in Visual Studio
I changed the setting in Visual Studio and it worked (due to the fact the correct nunit nuget packages were already added to the project).

Then came another problem while running the UITest.
It complained there was no ANDROID_HOME environment variable.
Fixed that by adding it.

Then when running the UITest, it complained about not finding the adb. Something todo with mismatching sdk versions. After a quick google I found this: [Getting your xamarin uitests to actually work](https://theconfuzedsourcecode.wordpress.com/2018/10/28/getting-your-xamarin-uitests-to-actually-work-not-thanks-to-xamarin-docs/
). So I downgraded the platform tools and it finally worked.
If that didn't worked, I would have just given up. It was just not worth the hassle.

If all those things were properly documented as possible pittfalls and available in the "getting started" section, it would help alot of starting contributers like me.
